models/articles.models.js

```diff
@@ -32,31 +32,66 @@ exports.selectArticleById = (id) => {
 };
 
 exports.fetchCommentsByArticleId = async (id) => {
-  try {
-    const existResult = await db.query({
-      text: "SELECT EXISTS (SELECT * FROM articles WHERE article_id = $1)",
-      values: [id],
-    });
-    if (!existResult.rows[0].exists) {
-      return Promise.reject({ status: 404, msg: `article_id ${id} not found` });
-    }
+  const existResult = await db.query({
+    text: "SELECT EXISTS (SELECT * FROM articles WHERE article_id = $1)",
+    values: [id],
+  });
+  if (!existResult.rows[0].exists) {
+    return Promise.reject({ status: 404, msg: `article_id ${id} not found` });
+  }
 
-    const result = await db.query({
-      text: `
+  const result = await db.query({
+    text: `
         SELECT * FROM comments
         WHERE article_id = $1
         ORDER BY created_at DESC;
         `,
-      values: [id],
+    values: [id],
+  });
+  if (result.rows.length === 0) {
+    return [];
+  }
+  return result.rows;
+};
+
+exports.postComment = async (id, author, comment) => {
+  if (!comment) {
+    return Promise.reject({ status: 422, msg: "Comment is required" });
+  }
+  if (!author) {
+    return Promise.reject({ status: 422, msg: "username is required" });
+  } else {
+    const existResult = await db.query({
+      text: "SELECT EXISTS (SELECT * FROM users WHERE username = $1)",
```

Comment:
Could you separate this out into a utility function for checking if the ID exists so you can reuse it where it's needed 💭